src/main/scala/com/codacy/tools/scala/seed/utils/ToolHelper.scala

```diff
@@ -7,9 +7,21 @@ object ToolHelper {
   implicit class ConfigurationOps(configuration: Option[List[Pattern.Definition]]) {
 
     def withDefaultParameters(implicit spec: Tool.Specification): Option[List[Pattern.Definition]] = {
-      configuration.map(
-        _.map(pattern => pattern.copy(parameters = pattern.parameters.orElse(defaultParameters(pattern.patternId))))
-      )
+      configuration.map(_.map { pattern =>
+        def toMap(setOpt: Option[Set[Parameter.Definition]]): Option[Map[Parameter.Name, Parameter.Definition]] =
+          setOpt.map(paramSet => paramSet.map(p => (p.name, p))(collection.breakOut))
+
+        val configurationParameters = toMap(pattern.parameters)
+        val defaults = toMap(defaultParameters(pattern.patternId))
+
+        val parameters =
+          configurationParameters
```

Comment:
we can probably rewrite this as something like:
```scala
defaults.filter(x => configurationParameters.contains(x.name == _.name)) ++ configurationParameters
```
if I"m not missing something

Reply:
This is not the same, we would not have the default values for parameters which are not in `configurationParameters`

Reply:
I'm convinced that this could be further simplified avoiding the usage of `toMap` and having the expected result, would you mind to add a test for the expected result so that I can validate my logic against it?